src/Compilers/Core/Portable/Compilation/OperationExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    internal interface IOperationSearchable
+    {
+        IEnumerable<IOperation> Descendants();
```

Comment:
Are these lazy?  If not, I think we standardized on ```ImmutableArray``` for list-like collections.  @AnthonyDGreen